### PR TITLE
fix(auth model): assignee missing from each type for roles

### PR DESCRIPTION
**Pull request by ale8k**
## Description

Assignee was missing. I've also moved the DSL around to it reads a bit more semantically.

## Engineering checklist
*Check only items that apply*

- [ ] Documentation updated
- [ ] Covered by unit tests
- [ ] Covered by integration tests

### CODE REVIEW

**Review by alesstimec — APPROVED** [review]
LGTM with an ask: could we please generate the JSON in the build process instead of commiting it every time?

**Comment by alesstimec** [comment]
**Note**: the entitlementsList in internal/jimmhttp/rebac_admin/entitlements.go will need to be fixed in a followup

**Comment by ale8k** [comment]
> LGTM with an ask: could we please generate the JSON in the build process instead of commiting it every time?

Not sure what you mean?

**Review by kian99 — APPROVED** [review]

**Review by kian99 — COMMENTED** [review]
I suggest when making quality-of-life improvements like re-ordering the auth model. Do the actual changes in one commit then the re-ordering in a separate commit. Makes it easier to review.

**Comment by ale8k** [comment]
> I suggest when making quality-of-life improvements like re-ordering the auth model. Do the actual changes in one commit then the re-ordering in a separate commit. Makes it easier to review.

To be fair, I thought this when I did it. It was more of a driveby

**Review by SimoneDutto — CHANGES_REQUESTED** [review]
please update the CI workflow here: https://github.com/canonical/jimm/blob/ab98ccc9a64b1fe6d72d305146b2fe6ad36b7d06/.github/workflows/ci.yaml#L31

The wrong auth model was caused because here we were not actually running the tests

Edit:
we are running the tests but the were lacking the role#assignee case.

File: openfga/tests.fga.yaml [hunk 1]
```diff
@@ -27,6 +27,40 @@ tuples:
...
+    - user: role:ro-role-2#assignee
+      relation: administrator
+      object: controller:ro-controller-1
+    - user: role:ro-role-2#assignee
+      relation: audit_log_viewer
+      object: controller:ro-controller-2
+
+    - user: role:ro-role-2#assignee
+      relation: administrator
+      object: model:ro-model-1
```
Comment (alesstimec) [inline openfga/tests.fga.yaml:39]: weren't we just complaining about confusing terminology? as i read these test i'm super confused by prefixes such as `user:gr-user-1` or `model:ro-model-1`..  i'd rather see we just used regular names, not names that "indicate" the purpose of the thing being named..

File: Makefile [hunk 2]
```diff
@@ -115,6 +115,8 @@ load-rock:
 	$(eval jimm_version := $(shell cat ./rocks/jimm.yaml | yq ".version"))
 	@sudo /snap/rockcraft/current/bin/skopeo --insecure-policy copy oci-archive:jimm_${jimm_version}_amd64.rock docker-daemon:jimm:latest
 
+transform-auth-model:
```
Comment (alesstimec) [inline Makefile]: suggestion: auth-model-json

File: openfga/README.md [hunk 3]
```diff
@@ -21,9 +21,7 @@ go install github.com/openfga/cli/cmd/fga@latest
 ## Adding / modifying [to] the authorsation model
 1. Open the authorisation_model.fga 
 2. Make your modification
-3. Open the Command Pallette using Ctrl+Shift+P (Windows) or Command+Shift+P (OSX)
-4. Select OpenFGA: Transform DSL to JSON
-5. Save the file over the existing authorisation_model.json
+3. Run: `make transform-auth-model`
```
Comment (alesstimec) [inline openfga/README.md:24]: much better.. but i would still prefer
`make auth-model-json`

**Review by alesstimec — APPROVED** [review]
LGTM with a few suggestions

File: openfga/tests.fga.yaml (hunk 1 above)
Comment (ale8k) [reply openfga/tests.fga.yaml:39]: I agree, but I followed the convention (and I get why piotr did it, to avoid collisions)

File: Makefile (hunk 2 above)
Comment (ale8k) [reply Makefile]: I'll leave this as it's following the verbs in the tool

**Review by SimoneDutto — APPROVED** [review]
thank you for updating the tests.

**Pull request merged**